internal/mode/static/usage/job_worker_test.go

```diff
@@ -0,0 +1,85 @@
+package usage_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+
+	"github.com/nginxinc/nginx-gateway-fabric/internal/mode/static/config"
+	"github.com/nginxinc/nginx-gateway-fabric/internal/mode/static/usage"
+	"github.com/nginxinc/nginx-gateway-fabric/internal/mode/static/usage/usagefakes"
+)
+
+func TestCreateUsageJobWorker(t *testing.T) {
```

Comment:
shall we also test for edge cases where data cannot be collected?

Reply:
We only log in that case, so we would be testing logs. Not sure how we feel about that type of unit test.

This would also require creating new interfaces and types for the collection, simply to test that we log errors properly. Doesn't seem worth it.